toolchain/sem_ir/node.h

```diff
@@ -92,7 +93,7 @@ constexpr BoolValue BoolValue::False = BoolValue(0);
 constexpr BoolValue BoolValue::True = BoolValue(1);
 
 // The ID of an integer literal.
-struct IntegerLiteralId : public IndexBase, public Printable<IntegerLiteralId> {
+struct IntegerValueId : public IndexBase, public Printable<IntegerValueId> {
```

Comment:
The comment and name are out of sync now.

I also feel like these aren't *quite* values -- they're closer to literals, no? They do the whole type-encodes-the-value thing?

Not sure what best to call them. They're not *lexical* literals, but "value" is getting fairly distinct meaning in the context of SemIR.

On the flip side, we use `BoolValue`, so there is also precedent.

Reply:
My point is more that they aren't going to represent all integer "value"s (as opposed to references etc) in the Carbon file, at least in the broad sense -- they're not the value model for an `i32`. And while we likely wouldn't call that just an "integer value" it seems like this could be confused with the representation for the value expression category of integers rather than the representation for a *specific* compile time integer's computed value.

Not sure what a better name for this construct is though. `CompTimeIntegerId` seems much too verbose if clear. I wish we had a word for the category of things between literals and (runtime) values...

Reply:
Mm, yeah, I think you've persuaded me. And in fact, I think my expectation was wrong:

While it's tempting to use this as the value model for compile-time `i32`s, we might want denser storage than an `APInt` for that case, and perhaps we shouldn't be using an `Id` mechanism for integer values that fit in 32 bits already. Even for `i128` compile-time values, it's not obvious to me that we'd want `APInt` storage rather than a dedicated table of `int128`s. So using this only for the type-encodes-the-value case might be best.

There's an even sharper distinction for `RealValueId`, where the type-encodes-the-value case wants an arbitrary rational number as the representation per the design, but once we go to `fN`, we presumably would want to represent the value as the bits of an `fN`.

So for now, perhaps the best thing to do is think of these as covering only the type-encodes-the-value cases. I'm not sure what that means for naming them, though.

Reply:
Renamed to `Real`, `Integer`, `RealId`, `IntegerId`. (Not renaming to `Rational` yet because we're not yet using a rational representation here, though "real" is even less accurate.)